File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
private:
    bool isvowel(char x)
    {
        return (x == 'a' or x == 'e' or x == 'i' or x == 'o' or x == 'u' or x == 'A' or
            x == 'E' or x == 'I' or x == 'O' or x == 'U');
    }
public:
    string reverseVowels(string s) 
    {
        int n = s.size();
        int i = 0 , j = n - 1;
        while(i < j)
        {
            if (isvowel(s[i]) and isvowel(s[j]))
            {
                swap(s[i],s[j]);
                i += 1;
                j -= 1;
            }
            else if (!isvowel(s[i]))
            {
                i += 1;
            }
            else if (!isvowel(s[j]))
            {
                j -= 1;
            }
        }
        return s;
    }
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        char = list(s)
        vowels = {'a','e','i','o','u','A','E','I','O','U'}
        left = 0
        right = len(s)-1
        while left<right:
            while left<right and char[left] not in vowels:
                left+=1
            while left<right and char[right] not in vowels:
                right-=1
            char[left],char[right] = char[right],char[left]
            left+=1
            right-=1
        return ''.join(char)
        